findutils: fix undeclared xargs dep on echo

abathur opened the pull request:
When invoked without a command parameter, xargs runs echo from PATH.
This change patches xargs to use a specific Nix-built echo.
###### Motivation for this change
Fix #89055

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

veprbl commented on pkgs/tools/misc/findutils/default.nix:
```diff
@@ -11,6 +11,10 @@ stdenv.mkDerivation rec {
     sha256 = "16kqz9yz98dasmj70jwf5py7jk558w96w0vgp3zf9xsqk3gzpzn5";
   };
 
+  prePatch = ''
```
```suggestion
  postPatch = ''
```

abathur replied:
the force-pushed change uses postPatch

veprbl reviewed (commented):
This will be a mass rebuild, so please target the staging branch.

abathur committed:
findutils: fix undeclared xargs dep on echo

When invoked without a command parameter, xargs runs echo from PATH.
This change patches xargs to use a specific Nix-built echo.

abathur commented:
> This will be a mass rebuild, so please target the staging branch.

I changed the target branch but it started listing other commits in addition to mine, so I've re-built the commit from staging and force-pushed. Hopefully that's fine.
